src/Simulation/QCTraceSimulator/SortedQubitPool.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime {
+
+    /// <summary>
+    /// Time to track qubit availability (DepthTime). Time is also used to track dependencies:
+    /// Gates that happen later on a qubit cannot be scheduled before gates on the same qubit.
+    /// This is tracked transitively via time, but zero gate depth prevents this from happening.
+    /// This is solved by having additional time component - DependencyTime.
+    /// </summary>
+    internal struct ComplexTime
+    {
+        /// <summary>
+        /// Time used to track qubit availability time. Double to match configuration interface.
```

Comment:
It's not clear what "availability time" is. Time when qubit was allocated? Time when the qubit was last used? Why qubit at all? The class doesn't seem to link to any particular qubit.

Reply:
Updated wording will be in the next push.
I thought about making this a general class, but it has too much ties with gate timing. Besides very few operations are defined to make it more general.